### DISCUSSION TITLE
Ability to use Postgres socket

### DISCUSSION THREAD

**Discussion by mahalel** [mealie-recipes/mealie#2460] [Feature Request]
Hi, would it be possible to add support to connect to Postgres via unix socket please?

The DSN would be in this format for Pydantic v1: `'postgresql://derp@/databasename?host=/var/run/postgresql'` 
I believe this is valid in pydantic v2 `postgresql://derp@%2Fvar%2Frun%2Fpostgresql/databasename`

Reading [db_providers.py](https://github.com/hay-kot/mealie/blob/4af9eec89dd0b309ebea752d715add3fe0980b3d/mealie/core/settings/db_providers.py#L46C4-L46C4) I don't believe I can achieve this DSN at the moment.

Thank you for a great project!

**Comment by apiontek** [comment]
I would also like to be able to do this. I already do it for a few other containers, by creating a shared docker volume mount, and it works great.

**Comment by boc-the-git** [comment]
Are there any functional benefits to this, or is it simply a different way of providing the configuration?

**Comment by apiontek** [comment]
My understanding is that UNIX sockets can be much faster than TCP sockets[^1][^2]

That said, it turns out this already works if one sets `POSTGRES_SERVER` to the socket path. I figured I would try it because this same approach has worked for some other apps & containers. For example, for me I set `POSTGRES_SERVER=/var/run/postgresql` instead of providing a docker container name/alias. (For reference, my postgres & mealie containers share volume assignment: `./sock/pg15:/var/run/postgresql`)

So the main thing missing would be a note in the documentation that a socket path can be set for that config value.

[^1]: [serverfault reference](https://serverfault.com/a/124518)
[^2]: [stackexchange reference](https://stackoverflow.com/a/15952170)
